File: training task/2_multiple_inheritance.cpp
#include <iostream>
#include <string>

using namespace std;
class Person {
protected:
    string name;
    int age;

public:
    Person(string name, int age) {
        this->name = name;
        this->age = age;
    }

    string getPersonDetails() const {
        return "Name: " + name + ", Age: " + to_string(age);
    }
};


class Academic {
protected:
    string institution;
    string major;

public:
    Academic(string institution, string major) {
        this->institution = institution;
        this->major = major;
    }

    
    string getAcademicDetails() const {
        return "Institution: " + institution + ", Major: " + major;
    }
};


class Student : public Person, public Academic {
private:
    string student_id;

public:

    Student(string name, int age, string institution, string major, string student_id)
        : Person(name, age), Academic(institution, major) {
        this->student_id = student_id;
    }

    string getStudentDetails() const {
        string person_details =getPersonDetails();
        string academic_details = getAcademicDetails();
        return person_details + ", " + academic_details + ", Student ID: " + student_id;
    }
};

int main() {
    Student student("Mou", 20, "NWU", "Computer Science", "S12345");
    cout << student.getStudentDetails() << endl;

    return 0;
}

File: training task/2_multiple_inheritance.py
class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def get_person_details(self):
        return f"Name: {self.name}, Age: {self.age}"

class Academic:
    def __init__(self, institution, major):
        self.institution = institution
        self.major = major

    def get_academic_details(self):
        return f"Institution: {self.institution}, Major: {self.major}"

class Student(Person, Academic):
    def __init__(self, name, age, institution, major, student_id):
        Person.__init__(self, name, age)
        Academic.__init__(self, institution, major)
        self.student_id = student_id

    def get_student_details(self):
        person_details = self.get_person_details()
        academic_details = self.get_academic_details()
        return f"{person_details}, {academic_details}, Student ID: {self.student_id}"


student = Student("Mou", 20, "NWU", "Computer Science", "S12345")
print(student.get_student_details())
